tools/js_optimizer.py

```diff
@@ -115,37 +93,49 @@ def create_optimizer_cmake():
         if WINDOWS:
           # Poor man's check for whether or not we should attempt 64 bit build
           if os.environ.get('ProgramFiles(x86)'):
-            cmake_generators = ['Visual Studio 12 Win64', 'Visual Studio 12', 'Visual Studio 11 Win64', 'Visual Studio 11', 'MinGW Makefiles', 'Unix Makefiles']
+            cmake_generators = [
+              'Visual Studio 15 2017 Win64', # The year component is mandatory for Visual Studio 2015 and above.
```

Comment:
Though not really :smile: https://gitlab.kitware.com/cmake/cmake/blob/0b33aee48b1f4027fe6086b5610434a5f29b0b8a/Source/cmGlobalVisualStudio15Generator.cxx#L16

Reply:
Thanks for pointing that out! The feature is mentioned in the documentation of VS2013 generator, but seems undocumented for VS2015, so I didn't really have a try...